From 33cd52312ef8a3abbb0709513f23b369941cb716 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 17 Feb 2016 15:48:04 +0100
Subject: [PATCH] Fix unique app handling

- Define variables explicitely for vala compiler.
- Use proper unique code for gtk3.
---
 Makefile.am           |  4 ++++
 lxclipboard/main.vala | 11 +++++++++--
 lxpolkit/main.vala    | 11 +++++++++--
 3 files changed, 22 insertions(+), 4 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index c9546f5..6d32d95 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -83,7 +83,9 @@ lxclipboard_lxclipboard_LDADD = \
     $(NULL)
 
 if USE_GTK3
+lxclipboard_lxclipboard_VALAFLAGS += --define USE_GTK3
 else
+lxclipboard_lxclipboard_VALAFLAGS += --define USE_GTK2
 lxclipboard_lxclipboard_VALAFLAGS += --pkg unique-1.0
 lxclipboard_lxclipboard_CPPFLAGS += $(UNIQUE_CFLAGS)
 lxclipboard_lxclipboard_LDADD += $(UNIQUE_LIBS)
@@ -128,7 +130,9 @@ lxpolkit_lxpolkit_LDADD = \
     $(NULL)
 
 if USE_GTK3
+lxpolkit_lxpolkit_VALAFLAGS += --define USE_GTK3
 else
+lxpolkit_lxpolkit_VALAFLAGS += --define USE_GTK2
 lxpolkit_lxpolkit_VALAFLAGS += --pkg unique-1.0
 lxpolkit_lxpolkit_CPPFLAGS += $(UNIQUE_CFLAGS)
 lxpolkit_lxpolkit_LDADD += $(UNIQUE_LIBS)
diff --git a/lxclipboard/main.vala b/lxclipboard/main.vala
index 059766b..7f0302e 100644
--- a/lxclipboard/main.vala
+++ b/lxclipboard/main.vala
@@ -37,8 +37,15 @@ namespace Lxsession
             }
 #endif
 # if USE_GTK3
-            Application app = new Application ();
-            app.run (args);
+            Gtk.Application app = new Gtk.Application (
+                "org.lxde.lxclipboard",
+                GLib.ApplicationFlags.FLAGS_NONE);
+            app.register ();
+
+            if(app.is_remote)
+            {
+                return 0;
+            }
 #endif
 
             clipboard_start ();
diff --git a/lxpolkit/main.vala b/lxpolkit/main.vala
index 8828e3a..56be273 100644
--- a/lxpolkit/main.vala
+++ b/lxpolkit/main.vala
@@ -42,8 +42,15 @@ namespace Lxsession
             }
 #endif
 # if USE_GTK3
-            Application app = new Application ();
-            app.run (args);
+            Gtk.Application app = new Gtk.Application (
+                "org.lxde.lxpolkit",
+                GLib.ApplicationFlags.FLAGS_NONE);
+            app.register ();
+
+            if(app.is_remote)
+            {
+                return 0;
+            }
 #endif
 
             policykit_agent_init();
-- 
2.7.1

